e3sm_diags/driver/lat_lon_driver.py

```diff
@@ -4,55 +4,110 @@
 import os
 
 import cdms2
-import MV2
 
 import e3sm_diags
 from e3sm_diags.driver import utils
 from e3sm_diags.logger import custom_logger
-from e3sm_diags.metrics import corr, max_cdms, mean, min_cdms, rmse, std
+from e3sm_diags.metrics import corr, mean, rmse, std
 from e3sm_diags.plot import plot
 
 logger = custom_logger(__name__)
 
 
+def create_and_save_data_and_metrics(parameter, mv1_domain, mv2_domain):
+    if parameter.ref_name != "":
+        # Regrid towards the lower resolution of the two
+        # variables for calculating the difference.
+        mv1_reg, mv2_reg = utils.general.regrid_to_lower_res(
+            mv1_domain,
+            mv2_domain,
+            parameter.regrid_tool,
+            parameter.regrid_method,
+        )
+
+        diff = mv1_reg - mv2_reg
+    else:
+        mv2_domain = None
+        mv2_reg = None
+        mv1_reg = mv1_domain
+        diff = None
+
+    metrics_dict = create_metrics(mv2_domain, mv1_domain, mv2_reg, mv1_reg, diff)
+
+    # Saving the metrics as a json.
+    metrics_dict["unit"] = mv1_domain.units
+
+    fnm = os.path.join(
+        utils.general.get_output_dir(parameter.current_set, parameter),
+        parameter.output_file + ".json",
+    )
+    with open(fnm, "w") as outfile:
+        json.dump(metrics_dict, outfile)
+
+    logger.info(f"Metrics saved in {fnm}")
+
+    plot(
+        parameter.current_set,
+        mv2_domain,
+        mv1_domain,
+        diff,
+        metrics_dict,
+        parameter,
+    )
+    utils.general.save_ncfiles(
+        parameter.current_set,
+        mv1_domain,
+        mv2_domain,
+        diff,
+        parameter,
+    )
+
+
 def create_metrics(ref, test, ref_regrid, test_regrid, diff):
     """Creates the mean, max, min, rmse, corr in a dictionary"""
+    # For input None, metrics are instantiated to 999.999.
+    # Apply float() to make sure the elements in metrics_dict are Jason serializable, i.e. np.float64 type is JSON serializable, but not np.float32.
```

Comment:
```suggestion
    # Apply float() to make sure the elements in metrics_dict are JSON serializable, i.e. np.float64 type is JSON serializable, but not np.float32.
```

Reply:
I think you mean JSON and not Jason haha. I committed the change for you.

Reply:
Thanks for catching this...